x/market/keeper/keeper.go

```diff
@@ -417,6 +419,61 @@ func (k *Keeper) CancelReplaceOrder(ctx sdk.Context, newOrder types.Order, origC
 	return &sdk.Result{Events: evts}, nil
 }
 
+func (k *Keeper) CancelReplaceMarketOrder(ctx sdk.Context, msg *types.MsgCancelReplaceMarketOrder) (*sdk.Result, error) {
+	// Use a fixed gas amount
+	ctx.GasMeter().ConsumeGas(gasPriceCancelReplaceOrder, "CancelReplaceMarketOrder")
```

Comment:
This function has quite a lot in common with `CancelReplaceLimitOrder`, ie. checking preconditions, finding original order, verifying that instruments remain the same. Would it be possible to capture that common logic in an internal, shared function?